### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by fponticelli**
* moved source code to `src/`
  * added `test/` for tests
  * added `build.hxml` to compile and run tests in neko (requires a local folder bin)
  * fixed API issue in `ContentSink`

### CODE REVIEW

**Comment by fponticelli** [comment]
Let me know if this works for you. I think it rationalize the project a little better. I also changed `haxelib.json` to use `src` as the entry point for the lib.

File: src/dom4/utils/Parser.hx [hunk 1]
```diff
@@ -1,24 +1,41 @@
...
+ * of those above. If you wish to allow use of your version of this file only
+ * under the terms of either the GPL or the LGPL, and not to allow others to
+ * use your version of this file under the terms of the MPL, indicate your
+ * decision by deleting the provisions above and replace them with the notice
+ * and other provisions required by the GPL or the LGPL. If you do not delete
+ * the provisions above, a recipient may use your version of this file under
+ * the terms of any one of the MPL, the GPL or the LGPL.
+ *
+ * ***** END LICENSE BLOCK ***** */
+
```
Comment (therealglazou) [inline src/dom4/utils/Parser.hx]: No, please, you cannot change the license block here. The doParse() method is adapted from code from the Haxe Foundation so the license comment block has to remain as is.

Comment (fponticelli) [inline src/dom4/utils/Parser.hx]: Sorry, I thought it was not intentional. I will fix it.

**Comment by fponticelli** [comment]
it should be fixed now

**Comment by fponticelli** [comment]
I should have put the `ping` here :)

**Pull request merged**